src/state.rs

```diff
@@ -0,0 +1,326 @@
+#![deny(missing_docs)]
+
+//! This module implements an abstraction for the Lurk state, which changes as
+//! Lurk code is evaluated
+
+use anyhow::{bail, Result};
+use once_cell::sync::OnceCell;
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use crate::Symbol;
+
+use super::package::{Package, SymbolRef};
+
+/// Keeps track of the current package for symbol resolution when reading and printing
+pub struct State {
+    current_package: SymbolRef,
+    symbol_packages: HashMap<SymbolRef, Package>,
+}
+
+impl State {
+    /// Wraps a state with `Rc<RefCell<...>>`
+    #[inline]
+    pub fn mutable(self) -> Rc<RefCell<Self>> {
+        Rc::new(RefCell::new(self))
+    }
```

Comment:
I think this method should not exist, not only because its name is inaccurate, but because the use of an `Rc<RefCell<T>>` should be explicit and left to the caller.

Reply:
It's just a wrapper for convenience. How should it be called?